src/webots/wren/WbWrenCamera.cpp

```diff
@@ -425,13 +427,23 @@ void WbWrenCamera::render() {
     updatePostProcessingParameters(CAMERA_ORIENTATION_FRONT);
 
   // Depth information needs to be conserved for post-processing shaders
+  const char *materialName = NULL;
+  if (!mIsColor)
+    materialName = "encodeDepth";
+  else if (mType == 's')
+    materialName = "segmentation";
   wr_scene_enable_depth_reset(wr_scene_get_instance(), false);
-  wr_scene_render_to_viewports(wr_scene_get_instance(), numActiveViewports, mViewportsToRender,
-                               (mType != 'c') ? "encodeDepth" : NULL, true);
+  wr_scene_render_to_viewports(wr_scene_get_instance(), numActiveViewports, mViewportsToRender, materialName, true);
 
   if (mIsSpherical)
     applySphericalPostProcessingEffect();
-
+  else if (mUpdateTextureFormatEffect) {
+    WrPostProcessingEffectPass *updateTextureFormatPass =
+      wr_post_processing_effect_get_pass(mUpdateTextureFormatEffect, "PassThrough");
+    wr_post_processing_effect_pass_set_input_texture(updateTextureFormatPass, 0,
+                                                     WR_TEXTURE(wr_frame_buffer_get_output_texture(mResultFrameBuffer, 0)));
+    wr_post_processing_effect_apply(mUpdateTextureFormatEffect);
+  }
```

Comment:
That's the only part I am not sure to understand.
We now need a post-processing effect in any case, this is to make sure the texture format is correct in every case?

Reply:
The format of the "rendering" texture is different than the "output" texture (passed to the controller).
For all type of previously existing cameras (color, radar and lidar) there are post processing effect(s) that, among other things, save the "output" texture in the correct format.
The segmentation type is the only one that doesn't have post-processing effects.

The "rendering" texture is also used internally by Webots for example for the overlays etc.
So using the "rendering" texture only or the "output" texture only will create conflicts with the format of the other textures.

Reply:
Ok, that makes perfectly sense, thanks a lot for the explanation!